packages/react-components/react-utilities/src/compose/slot.ts

```diff
@@ -91,6 +91,16 @@ export function resolveShorthand<Props extends UnknownSlotProps | null | undefin
   ) {
     return { children: value } as Props;
   }
+  if (value && typeof value !== 'object' && process.env.NODE_ENV !== 'production') {
+    // eslint-disable-next-line no-console
+    console.error(
+      [
+        `[slot.resolveShorthand]: invalid slot value ${value}.\n`,
+        'A valid value is a slot shorthand or slot properties object.\n',
+        'A slot shorthand can be a string, number, array, or JSX.Element.',
```

Comment:
```suggestion
        'Slots accept strings, numbers, arrays and JSX elements a values.',
```

IMO it's clear enough